arcgis-ios-sdk-samples/Route and directions/Navigate route with rerouting/NavigateRouteWithReroutingViewController.swift

```diff
@@ -0,0 +1,364 @@
+        } else {
+            mapView.setViewpointGeometry(geometry) { [weak self] _ in
+                // Get the initial zoomed view point.
+                self?.defaultViewPoint = self?.mapView.currentViewpoint(with: .centerAndScale)
+            }
+        }
+    }
+    
+    // MARK: UI
+    
+    func setStatus(message: String) {
+        statusLabel.text = message
+    }
+    
+    // MARK: UIViewController
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Add the source code button item to the right of navigation bar.
+        (navigationItem.rightBarButtonItem as? SourceCodeBarButtonItem)?.filenames = ["NavigateRouteWithReroutingViewController"]
+        // Avoid the overlap between the status label and the map content.
+        mapView.contentInset.top = CGFloat(statusLabel.numberOfLines) * statusLabel.font.lineHeight
+        
+        // Solve the route as map loads.
+        routeTask.defaultRouteParameters { [weak self] (params: AGSRouteParameters?, error: Error?) in
+            guard let self = self else { return }
+            if let params = params {
+                // Explicitly set values for parameters.
+                params.returnDirections = true
+                params.returnStops = true
+                params.returnRoutes = true
+                params.outputSpatialReference = .wgs84()
+                params.setStops(self.makeStops())
+                self.routeParameters = params
+                self.routeTask.solveRoute(with: params) { [weak self] (result, error) in
+                    if let result = result {
+                        self?.route = result.routes.first
+                        self?.didSolveRoute(with: .success(result))
+                    } else if let error = error {
+                        self?.didSolveRoute(with: .failure(error))
+                    }
+                }
+            } else if let error = error {
+                self.presentAlert(error: error)
+                self.setStatus(message: "Failed to get route parameters.")
+            }
+        }
+    }
+}
+
+// MARK: - AGSRouteTrackerDelegate
+
+extension NavigateRouteWithReroutingViewController: AGSRouteTrackerDelegate {
+    func routeTracker(_ routeTracker: AGSRouteTracker, didGenerateNewVoiceGuidance voiceGuidance: AGSVoiceGuidance) {
+        setSpeakDirection(with: voiceGuidance.text)
+    }
+    
+    func routeTracker(_ routeTracker: AGSRouteTracker, didUpdate trackingStatus: AGSTrackingStatus) {
+        updateTrackingStatusDisplay(routeTracker: routeTracker, status: trackingStatus)
+    }
+    
+    func routeTracker(_ routeTracker: AGSRouteTracker, rerouteDidCompleteWith trackingStatus: AGSTrackingStatus?, error: Error?) {
+        if let error = error {
+            self.presentAlert(error: error)
+        } else {
+            // Get the new directions.
+            directionsList = (trackingStatus?.routeResult.routes.first!.directionManeuvers)!
+        }
```

Comment:
```suggestion
        if let error = error {
            presentAlert(error: error)
        } else if let status = trackingStatus {
            // Get the new directions.
            directionsList = status.routeResult.routes.first!.directionManeuvers
        }
```
consider doing sth like 👆